### PR TITLE
 Preserve monitor connections for any preserved vms across test runs

### PR DESCRIPTION
**Pull request by pevogam**
This is an addition to the original implementation of vm persistence
that preserves the vm objects, i.e. PR: Optional vm persistence #2105.

Signed-off-by: Plamen Dimitrov <plamen.dimitrov@intra2net.com>

### CODE REVIEW

**Comment by pevogam** [comment]
@luckyh Would you mind sharing your opinion here?

File: virttest/qemu_monitor.py [hunk 1]
```diff
@@ -492,7 +492,7 @@ def is_responsive(self):
         try:
             self.verify_responsive()
             return True
-        except MonitorError:
+        except (MonitorError, ValueError):
```
Comment (luckyh) [inline virttest/qemu_monitor.py:495]: Can we simply make it by checking the `fileno`, in this case?
```python
if self._socket.fileno() < 0:
    return False
```

Comment (pevogam) [reply virttest/qemu_monitor.py:495]: Good idea, will test and come back to you here.

Comment (pevogam) [reply virttest/qemu_monitor.py]: This suggestion works and is more refined, thanks! I pushed the changes and have tested them locally already.

**Review by luckyh — APPROVED** [review]
LGTM

**Comment by pevogam** [comment]
@luckyh I have rebased the branch on top of the most recent tag, all CI passes now and we have two acks.

**Comment by luckyh** [comment]
> @luckyh I have rebased the branch on top of the most recent tag, all CI passes now and we have two acks.

Thanks!

**Pull request merged**